server/controllers/articles.js

```diff
@@ -27,67 +27,59 @@ class ArticleController {
       // Validate the rating
       const articleDetails = await validateRating(req.body);
 
-      const { articleId, rate } = articleDetails;
+      const { rate } = articleDetails;
+      const { articleId } = req.params;
 
       // Check if the article exists
       const article = await Article.findOne({
-        where: { id: articleId }
-      });
-
-      if (!article) return Response.error(res, 404, 'Article does not exist');
-
-      // Get the full details of user and article
-      const moreDetails = await Article.findOne({
         where: { id: articleId },
-        attributes: { exclude: ['updatedAt', 'createdAt'] },
         include: [
           {
             model: User,
             as: 'author',
-            attributes: { exclude: ['password', 'updatedAt', 'createdAt'] }
+            attributes: [
+              'id',
+              'username',
+              'email',
+              'firstname',
+              'lastname',
+              'middlename',
+              'active'
+            ]
           }
-        ],
+        ]
       });
 
-      let fullRatingInfo = { article: moreDetails };
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      let fullRatingInfo = { article };
+      let statusCode = 201;
+      let result;
 
-      // Check if this user has made a rating before
-      const userArticleRating = await Rating.findOne({
-        where: {
-          userId,
-          articleId: req.body.articleId
-        }
+      const [newRating, created] = await Rating.findOrCreate({
```

Comment:
what do you think is the point of having `associations` in sequelize if you gonna be doing stuff like this?

article.`set/add/get/etc`ratings (one or more of these methods is available to you) USE THEM